linera-views/src/memory.rs

```diff
@@ -16,43 +17,18 @@ use thiserror::Error;
 pub type MemoryStoreMap = BTreeMap<Vec<u8>, Vec<u8>>;
 
 /// A virtual DB client where data are persisted in memory.
-pub type MemoryClient = Arc<RwLock<MutexGuardArc<MemoryStoreMap>>>;
-
-/// An implementation of [`crate::common::Context`] that stores all values in memory.
-pub type MemoryContext<E> = ContextFromDb<E, MemoryClient>;
-
-impl<E> MemoryContext<E> {
-    /// Creates a [`MemoryContext`].
-    pub fn new(guard: MutexGuardArc<MemoryStoreMap>, extra: E) -> Self {
-        Self {
-            db: Arc::new(RwLock::new(guard)),
-            base_key: Vec::new(),
-            extra,
-        }
-    }
-}
-
-/// Provides a `MemoryContext<()>` that can be used for tests.
-pub fn create_test_context() -> MemoryContext<()> {
-    let state = Arc::new(Mutex::new(BTreeMap::new()));
-    let guard = state
-        .try_lock_arc()
-        .expect("We should acquire the lock just after creating the object");
-    MemoryContext::new(guard, ())
-}
-
-/// Creates a test memory client.
-pub fn create_test_memory_client() -> MemoryClient {
-    let state = Arc::new(Mutex::new(BTreeMap::new()));
-    let guard = state
-        .try_lock_arc()
-        .expect("We should acquire the lock just after creating the object");
-    Arc::new(RwLock::new(guard))
-}
+pub type MemoryClientInternal = Arc<RwLock<MutexGuardArc<MemoryStoreMap>>>;
 
 #[async_trait]
-impl KeyValueStoreClient for MemoryClient {
+impl KeyValueStoreClient for MemoryClientInternal {
     const MAX_CONNECTIONS: usize = 1;
+    // For the memory container, memory is the limit and so usize::MAX
+    // could be used. But when memory is used for tests we want to exercise
+    // the splitting mechanism and so we set it artificially to 100.
+    #[cfg(any(test, feature = "test"))]
+    const MAX_VALUE_SIZE: usize = 100;
```

Comment:
This is not amazing because in a near future, users of a `KeyValueStoreClient` implementation should probably throw an error when a value size exceeds `MAX_VALUE_SIZE`.

I'm afraid the only solution here is to define a `TestMemoryClient` in `value_splitting.rs` and manually delegate the calls to the real one (except MAX_VALUE_SIZE obviously).

Reply:
Ok, this has been done and is now reasonably generic.